### PR TITLE
net-im/telegram-desktop: Bump to 4.10.3

### PR DESCRIPTION
**Pull request by mid-kid**
The old versions are currently utterly and completely broken, severely out of date, possibly riddled with security issues, and I don't wish to waste a second of my time to fix them. However, they're the only stable versions. Can I remove them?

Also, can I put telegram-desktop on a fast-track for stabilization? For the reasons above, it'd be nice to not have to wait a full month before requestion stabilization of the new package. Maybe a week or something would be more appropriate, and from there I can pick up the old upgrade schedule again.

@gyakovlev

### CODE REVIEW

File: net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+		>=dev-qt/qtgui-5.15:5[dbus=,jpeg,png,wayland?,X?]
+		>=dev-qt/qtimageformats-5.15:5
+		>=dev-qt/qtnetwork-5.15:5[ssl]
+		>=dev-qt/qtsvg-5.15:5
+		>=dev-qt/qtwidgets-5.15:5[png,X?]
+		kde-frameworks/kcoreaddons:5
+		webkit? ( >=dev-qt/qtdeclarative-5.15:5 )
+	)
+	qt6? (
+		>=dev-qt/qtbase-6.5:6[dbus=,gui,network,opengl,wayland?,widgets,X?]=
```
Comment (joecool1029) [inline net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild:67]: Drop the = at end to fix pkgcheck, should update metadata.xml with current telegram useflags as well.

**Review by joecool1029 — CHANGES_REQUESTED** [review]

**Comment by Kangie** [comment]
> However, they're the only stable versions. Can I remove them?

No, we need to avoid dropping stable keywords where possible. We can request a security stablisation for these early by filing an appropriate bug and blocking the libwebp tracker: https://bugs.gentoo.org/914072

Here's an example for chromium: https://bugs.gentoo.org/915151

File: net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild (hunk 1 above)
Comment (mid-kid) [reply net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild:67]: The `=` is required (and is a valid operator as far as I'm aware). You cannot build build telegram without Qt6DBus when dev-qt/qtbase was built with USE=dbus, as Qt6Gui will depend on Qt6DBus then.

**Comment by mid-kid** [comment]
@Kangie Thanks for the insight. Nothing better than abusing security concerns as an excuse to push things forward. :)

File: net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild (hunk 1 above)
Comment (ionenwks) [reply net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild:67]: In qt's own packages, sometime had to resort to sed instead given disabling sub-component of qtbase tend to end badly.

Comment (mid-kid) [reply net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild:67]: @joecool1029 Oh right, I always get the position of the `=` wrong. It should be before the `[`

@ionenwks You're right. I should rethink how I control these dependencies. I don't want to force wayland but I don't like having to `sed` my way around everything or have bespoke patches...

Comment (ionenwks) [reply net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild]: The others are pretty safe, aka wayland's diable_find_package which is not part of qtbase is fine. It's when you disable bits of qtbase (Gui, Widgets, DBus, Opengl, etc...) that you end up breaking assumptions. So it'd be more an exception that a need to change everything. 

On a side-note, disabling Quick will force disable QuickWidgets in a similar way (aka QuickWidgets tries to find Quick, then fails). So specifying both is not strictly needed.

Comment (mid-kid) [reply net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild]: I prefer both as it helps me keep track of the automagic deps I've handled :)
Thanks a lot for the comment, I guess I can handle the DBus dependency through `sed` and the rest through find_package disabling.

**Comment by mid-kid** [comment]
Done! Feel free to merge whenever.

**Comment by mid-kid** [comment]
Ok but for real this time
@gyakovlev @ionenwks

File: net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+		-print0 | xargs -0 sed -i \
+		-e '/pkg_check_modules(/s/[^ ]*)/REQUIRED &/' \
+		-e '/find_package(/s/)/ REQUIRED)/' || die
+	# Make sure to check the excluded files for new
+	# CMAKE_DISABLE_FIND_PACKAGE entries.
+
+	# Control QtDBus dependency from here, to avoid messing with QtGui.
+	if ! use dbus; then
+		sed -e '/find_package(Qt[^ ]* OPTIONAL_COMPONENTS/s/DBus *//' \
+			-i cmake/external/qt/package.cmake
```
Comment (dlan17) [inline net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild:151]: sed could fail if file not found, so missing "|| die" ?

Comment (mid-kid) [reply net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild:151]: You're right. Thanks!

**Review by ionenwks — APPROVED** [review]
> Ok but for real this time gyakovlev ionenwks

I assume(?) gyakovlev is still mostly busy. Haven't given this a real review, nor did extensive build tests (just tried a basic build with qt6, and I don't use telegram myself), but no harm in just quick merging it considering state of in-tree versions.

Thanks for handling this.

Also, it did fail with:
```
ninja: error: 'Telegram/lib_base/CppGir::cppgir', needed by 'gen/lib_base_Mpris_cppgir.timestamp', missing and no known rule to make it
```
Apparently caused by the old cppgir and it did not upgrade it first. So I'll add a `>=0_p20230926` check before merging.

**By the way**, I think(?) it's broken with (unkeyworded) qt6.6.0-rc, I got:
```
/var/tmp/portage/net-im/telegram-desktop-4.10.3/work/tdesktop-4.10.3-full/Telegram/SourceFiles/boxes/peers/edit_participant_box.cpp:404:35: error: ambiguous overload for 'operator=' (operand types are 'QPointer<Ui::BoxContent>' and 'QPointer<Ui::GenericBox>')
  404 |                                 }));
```
Did not look in-depth but built fine after I downgraded to 6.5.3. 6.6.0 is due to release in 3 days (Oct 10th) or around there assuming it's not delayed.

**Comment by mid-kid** [comment]
Oh boy, I guess that's why I should also test on unstable rather than stable only...
Am I allowed to just require <=6.5 for now? I doubt it'll take long to be solved upstream.

**Comment by ionenwks** [comment]
> Oh boy, I guess that's why I should also test on unstable rather than stable only... Am I allowed to just require <=6.5 for now? I doubt it'll take long to be solved upstream.

Sure, as long as it's not something that drags.

**Pull request closed without merging**